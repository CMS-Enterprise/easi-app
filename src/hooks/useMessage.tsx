import React, {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useState
} from 'react';
import { useLocation } from 'react-router-dom';

import Alert, { AlertProps } from 'components/shared/Alert';

type MessageProps = {
  message?: string | React.ReactElement;
  alertProps?: Omit<AlertProps, 'children'>;
};

type SetMessage = (
  message?: string | React.ReactElement,
  alertProps?: Omit<AlertProps, 'children'>
) => void;

const MessageContext = createContext<
  | {
      Message: ({ className }: { className?: string }) => JSX.Element | null;
      showMessage: SetMessage;
      showMessageOnNextPage: SetMessage;
    }
  | undefined
>(undefined);

// MessageProvider manages the state necessary for child components to
// use the useMessage hook.
const MessageProvider = ({ children }: { children: ReactNode }) => {
  const [queuedMessage, setQueuedMessage] = useState<MessageProps>({});
  const [message, setMessage] = useState<MessageProps>({});

  const location = useLocation();

  const [lastPathname, setLastPathname] = useState(location.pathname);

  const Message = (props: { className?: string }) => {
    // If no message, return null
    if (!message.message) return null;

    // If message has alert props, wrap in component
    if (message.alertProps) {
      return (
        <Alert {...message.alertProps} {...props}>
          {message.message}
        </Alert>
      );
    }

    // Return plain message
    return <>{message.message}</>;
  };

  useEffect(() => {
    if (lastPathname !== location.pathname) {
      setMessage(queuedMessage);
      setQueuedMessage({});
      setLastPathname(location.pathname);
    }
  }, [message, queuedMessage, lastPathname, location.pathname]);

  return (
    <MessageContext.Provider
      value={{
        Message,
        showMessage: (value, alertProps) =>
          setMessage({
            message: value,
            alertProps
          }),
        showMessageOnNextPage: (value, alertProps) =>
          setQueuedMessage({
            message: value,
            alertProps
          })
      }}
    >
      {children}
    </MessageContext.Provider>
  );
};

// useMessage is a hook that provides the ability to show notifications to the
// user. Components will use the hook to get access to two methods:
//
// showMessage: display a message to the user immediately. This is useful
// when an interaction won't cause an immediate redirect to a new page.
//
// showMessageOnNextPage: schedule a message to be shown on the next
// visited by the user. Pages are identified using their URL and therefore
// messages won't appear until the URL changes.
//
// Currently there is only support for showing a single message.
const useMessage = () => {
  const context = useContext(MessageContext);
  if (context === undefined) {
    throw new Error('useMessage must be used within a MessageProvider');
  }

  return context;
};

export default useMessage;
export { useMessage, MessageProvider };
