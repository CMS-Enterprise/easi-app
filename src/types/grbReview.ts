import {
  GetGRBReviewersComparisonsQuery,
  SystemIntakeGRBReviewerRole,
  SystemIntakeGRBReviewerVotingRole
} from 'gql/gen/graphql';

export type GRBReviewerFields = {
  userAccount: {
    username: string;
    commonName: string;
    email: string;
  };
  votingRole: SystemIntakeGRBReviewerVotingRole;
  grbRole: SystemIntakeGRBReviewerRole;
};

export type GRBReviewerComparisonIntake =
  GetGRBReviewersComparisonsQuery['compareGRBReviewersByIntakeID'];

export type GRBReviewerComparison =
  GRBReviewerComparisonIntake[number]['reviewers'][number];
