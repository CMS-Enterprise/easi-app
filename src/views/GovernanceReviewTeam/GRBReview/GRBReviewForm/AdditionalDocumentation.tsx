import React from 'react';

import { EasiFormProvider, useEasiForm } from 'components/EasiForm';

import GRBReviewFormStepWrapper from './GRBReviewFormStepWrapper';

// TODO: Update fields type
type AdditionalDocumentationFields = {};

const AdditionalDocumentation = () => {
  const form = useEasiForm<AdditionalDocumentationFields>();

  return (
    <EasiFormProvider<AdditionalDocumentationFields> {...form}>
      <GRBReviewFormStepWrapper>
        <p>Fields here</p>
      </GRBReviewFormStepWrapper>
    </EasiFormProvider>
  );
};

export default AdditionalDocumentation;
