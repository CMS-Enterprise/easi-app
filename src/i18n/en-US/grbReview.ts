import {
  SystemIntakeGRBReviewerRole,
  SystemIntakeGRBReviewerVotingRole
} from 'gql/gen/graphql';

import { GRBReviewFormAction } from 'types/grbReview';
import { Translation } from 'types/util';

export const grbReviewerVotingRolesTranslation: Translation<SystemIntakeGRBReviewerVotingRole> =
  {
    VOTING: 'Voting',
    NON_VOTING: 'Non-voting',
    ALTERNATE: 'Alternate'
  };

export const grbReviewerRolesTranslation: Translation<SystemIntakeGRBReviewerRole> =
  {
    CO_CHAIR_CIO: 'Co-Chair - CIO',
    CO_CHAIR_CFO: 'CO-Chair - CFO',
    CO_CHAIR_HCA: 'CO-Chair - HCA',
    ACA_3021_REP: 'ACA 3021 Rep',
    CCIIO_REP: 'CCIIO Rep',
    PROGRAM_OPERATIONS_BDG_CHAIR: 'Program Operations BDG Chair',
    CMCS_REP: 'CMCS Rep',
    FED_ADMIN_BDG_CHAIR: 'Fed Admin BDG Chair',
    PROGRAM_INTEGRITY_BDG_CHAIR: 'Program Integrity BDG Chair',
    QIO_REP: 'QIO Rep',
    SUBJECT_MATTER_EXPERT: 'Subject Matter Expert (SME)',
    OTHER: 'Other'
  };

const messages: {
  error: Translation<GRBReviewFormAction>;
  success: Translation<GRBReviewFormAction>;
} = {
  error: {
    add: 'There was an issue adding GRB reviewers. Please try again, and if the error persists, try again at a later date.',
    edit: 'There was an issue updating roles for this GRB reviewer. Please try again, and if the error persists, try again at a later date.',
    remove:
      'There was an issue removing this GRB reviewer. Please try again, and if the error persists, try again at a later date.'
  },
  success: {
    add: 'You added <strong>1 reviewer</strong> to this GRB review.',
    add_plural:
      'You added <strong>{{count}} reviewers</strong> to this GRB review.',
    edit: 'You updated roles for {{commonName}} for this GRB review.',
    remove: 'You removed <strong>{{commonName}}</strong> as a GRB reviewer.'
  }
};

export default {
  title: 'GRB review',
  description:
    'Many projects coming through the IT Governance process require review by the Governance Review Board (GRB). These reviews may happen at regularly scheduled GRB meetings or during an alternate asynchronous review process. GRB members will review project details, discuss any concerns, and vote on the merits of the project under consideration. ',
  featureInProgress: 'Feature in progress',
  featureInProgressText:
    'The capabilities currently available allow Governance Admin Team members to invite GRB reviewers to access documentation related to requests. Future capabilities will include voting, discussions with project teams, timelines, and more. Currently, however, all communication with the project team must be completed via email or during meetings. If you have any questions, please complete the <a>feedback form</a> in the footer of this page.',
  startGrbReview: 'Start GRB review',
  reviewStartedOn: 'Review started on {{date}}',
  startReviewModal: {
    heading: 'Start this GRB review?',
    text: 'Starting this review will send email notifications to {{count}} GRB reviewers. Any reviewer you add after starting this review will automatically receive a notification.',
    startReview: 'Start review and send notifications'
  },
  startGrbReviewError:
    'There was an issue starting this GRB review. Please try again, and if the error persists, try again at a later date.',
  participantsStartReviewAlert:
    'Adding GRB reviewers will not send them an invitation until you start the GRB review using the <a>Start GRB Review</a> button above, though they will still be able to access content if they sign into EASi.',
  participants: 'Participants',
  participantsText:
    'The Governance Admin Team has invited you to view documentation related to this IT Governance request. Use the navigation to the left or the buttons below to view the documentation.',
  availableDocumentation: 'Available documentation',
  viewBusinessCase: 'View Business Case',
  viewIntakeRequest: 'View Intake Request',
  viewOtherDocuments: 'View other supporting documents',
  addGrbReviewer: 'Add a GRB reviewer',
  reviewDetails: {
    title: 'Review details',
    text: 'Additional content relevant to this GRB review.',
    grbFeedback: {
      title: 'GRT recommendations to the GRB',
      text: 'The Governance Review Team (GRT) has provided recommendations and feedback about this project.',
      emptyAlert:
        'The Governance Review Team (GRT) did not provide recommendations or feedback for this project.',
      show: 'Show GRT recommendations',
      hide: 'Hide GRT recommendations'
    }
  },
  supportingDocuments: 'Supporting documents',
  supportingDocumentsText:
    'The documents below will help the GRB review this IT Governance request, and were completed during the course of this IT Governance request or were added by the requester and/or the Governance Admin Team. You may add additional documents and may remove any that have been added by Governance Admin Team members.',
  businessCaseOverview: {
    title: 'Business Case overview',
    submitted: 'Last Updated',
    need: 'Business or user need',
    preferredSolution: 'Preferred solution: Summary',
    noSolution: 'No answer added.',
    linkToBusinessCase: 'View full Business Case',
    unsubmittedAlertText: 'There is not yet a Business Case for this request.'
  },
  additionalDocuments: 'Additional documents',
  additionalDocsLink: 'Add another supporting document',
  documentsIntakeLinkTitle: 'EASi Intake Request form',
  documentsIntakeLinkText: 'View in EASi',
  documentsIntakeSubmitted: 'submitted',
  participantsTable: {
    name: 'Name',
    votingRole: 'Voting role',
    grbRole: 'GRB role',
    actions: 'Actions',
    noReviewers:
      'You have not yet added GRB reviewers. Add reviewers using the button above.'
  },
  addAnotherGrbReviewer: 'Add another GRB reviewer',
  closedRequest:
    'This request is closed. Please <a>re-open</a> it before adding additional reviewers',
  form: {
    title: 'Add a GRB reviewer',
    description:
      'Add details about this Governance Review Board (GRB) reviewer, including their name, voting role, and GRB role such as the Budget Development Group (BDG) they represent.',
    returnToRequest_edit: 'Don’t save and return to request details',
    returnToRequest_add: 'Don’t add and return to request details',
    addViaEUA: 'Add via EUA',
    addFromRequest: 'Add from previous request',
    addFromRequestDescription:
      'EASi allows you to bulk-add GRB reviewers that were previously added to past GRB reviews.',
    itGovernanceRequests: 'EASi IT Governance requests',
    itGovernanceRequestsHelpText:
      'Search this list of previous Governance requests. Any requests without assigned GRB reviewers are filtered out of this list.',
    selectRequestAlert:
      'Select a request above to populate a list of possible GRB reviewers to select from.',
    grbReviewerResults:
      '<strong>{{count}} results.</strong> Select reviewers you wish to add to your current request’s GRB review. You may edit roles after adding reviewers. Checkboxes are disabled for individuals who have already been added to this review.',
    grbMemberName: 'GRB member name',
    grbMemberNameHelpText: 'This field searches CMS’ EUA database.',
    duplicateReviewerAlert:
      "This individual is already listed as a GRB reviewer for this request. Please choose a different reviewer or return to the main review screen to manage this reviewer's roles.",
    votingRole: 'Voting role',
    votingRolesInfo: {
      label: 'What voting roles are available?',
      items: [
        '<dt>Voting:</dt> <dd>Assign this role to any voting members of the GRB for standard GRB reviews, or to the GRB co-chairs for a waiver or expedited review. Individuals with this role will vote on the merit of this project.</dd>',
        '<dt>Alternate:</dt> <dd>Assign this role to any backup voting members for standard GRB reviews.</dd>',
        '<dt>Non-voting:</dt> <dd>Assign this role to any individual who should be able to see the review content but will not vote on the merit of the project.</dd>'
      ]
    },
    grbRole: 'GRB role',
    grbRoleHelpText:
      'Select the role that this reviewer has on the GRB, such as the BDG that they represent.',
    submit_add: 'Add reviewer',
    submit_add_zero: 'Add reviewer',
    submit_add_plural: 'Add reviewers ({{count}})',
    submit_edit: 'Save changes',
    removeGrbReviewer: 'Remove GRB reviewer',
    infoAlertReviewNotStarted:
      'Adding GRB reviewers will not send them an invitation until you start the GRB review, though they will still be able to access content if they sign into EASi. This individual will be able to see information about this IT Governance request including the Intake Request form, Business Case, and other supporting documents. Please make sure this individual should be able to access this information before you proceed. They will not be able to take any actions on the request or see Admin notes.',
    infoAlertReviewStarted:
      'Adding a reviewer will send them an informational notification email with a link to EASi. This individual will be able to see information about this IT Governance request including the Intake Request form, Business Case, and other supporting documents. Please make sure this individual should be able to access this information before you proceed. They will not be able to take any actions on the request or see Admin notes.'
  },
  messages,
  votingRoles: grbReviewerVotingRolesTranslation,
  reviewerRoles: grbReviewerRolesTranslation,
  removeModal: {
    title: 'Remove {{commonName}} as a GRB reviewer?',
    text: 'Removing this reviewer will remove their access to documentation about this request. You may add them again at a later date if necessary.',
    remove: 'Remove reviewer'
  },
  homepage: {
    participationNeeded: 'GRB participation needed',
    participationNeededText:
      'The Governance Admin Team asks that the Governance Review Board (GRB) review and vote on the merits of IT projects. Depending on the project and timeframe, this may done asynchronously or during a regular GRB review meeting. Use the button below to access details for IT Governance requests that require a GRB review. ',
    showGrbReviews: 'Show GRB reviews',
    hideGrbReviews: 'Hide GRB reviews',
    grbDate: 'GRB date',
    noDateSet: 'No date set'
  },
  adminTask: {
    title: 'Admin Task',
    setUpGRBReview: {
      title: 'Set up GRB review',
      description:
        'EASi will guide you through setting up a GRB review, including the decision between an asynchronous review in EASi and a standard review meeting.',
      whatDoINeedLabel: 'What do I need in order to set up a GRB review?',
      whatDoINeed: [
        '<bold>Presentation recording and transcript</bold>: Organize a time to asynchronously record this project team’s presentation. You will need to add a link to the recording during the setup process.',
        '<bold>Additional supporting documents</bold>: EASi will automatically add a reference to the Business Case, Intake Request form, and any supporting documents already uploaded by the requester. Please have on hand any additional documents for the GRB to consider as a part of this request.',
        '<bold>GRB reviewers</bold>: EASi will guide you through the process of adding voting and non-voting GRB reviewers. Please have on hand the names, email addresses, or EUA IDs of these individuals.',
        '<bold>Timeframe</bold>: Decide on the number of days allotted for this GRB review. You will be able to add more time during the review process if needed.'
      ]
    },
    sendReviewReminder: {
      title: 'Send review reminder',
      description:
        'GRB voting members will get an automatic reminder when there is one day left in the review process. You may use the button below to send an additional reminder at any time.',
      sendReminder: 'Send reminder'
    },
    takeADifferentAction: 'or, take a different action'
  }
};
