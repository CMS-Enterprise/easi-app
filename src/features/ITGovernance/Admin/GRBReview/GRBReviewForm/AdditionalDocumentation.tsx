import React from 'react';

import { EasiFormProvider, useEasiForm } from 'components/EasiForm';
import { GRBReviewFormStepProps } from 'types/grbReview';

import GRBReviewFormStepWrapper from './GRBReviewFormStepWrapper';

// TODO: Update fields type
type AdditionalDocumentationFields = {};

const AdditionalDocumentation = ({ grbReview }: GRBReviewFormStepProps) => {
  const form = useEasiForm<AdditionalDocumentationFields>();

  return (
    <EasiFormProvider<AdditionalDocumentationFields> {...form}>
      <GRBReviewFormStepWrapper grbReview={grbReview}>
        <p>Fields here</p>
      </GRBReviewFormStepWrapper>
    </EasiFormProvider>
  );
};

export default AdditionalDocumentation;
