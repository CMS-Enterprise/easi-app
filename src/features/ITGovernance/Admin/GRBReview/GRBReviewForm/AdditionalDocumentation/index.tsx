import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { Button } from '@trussworks/react-uswds';
import DocumentsTable from 'features/ITGovernance/_components/DocumentsTable';
import { SystemIntakeGRBReviewType } from 'gql/generated/graphql';

import Alert from 'components/Alert';
import CollapsableLink from 'components/CollapsableLink';
import { GRBReviewFormStepProps } from 'types/grbReview';

import GRBReviewFormStepWrapper from '../GRBReviewFormStepWrapper';

const AdditionalDocumentation = ({ grbReview }: GRBReviewFormStepProps) => {
  const { t } = useTranslation('grbReview');
  const history = useHistory();

  return (
    <GRBReviewFormStepWrapper grbReview={grbReview} requiredFields={false}>
      <CollapsableLink
        id="documentsList"
        className="margin-top-3 padding-y-0"
        label={t('setUpGrbReviewForm.documents.collapsableText.label')}
      >
        <p className="line-height-body-5 margin-y-0 tablet:grid-col-9">
          {t('setUpGrbReviewForm.documents.collapsableText.description')}
        </p>

        <ul className="line-height-body-5 margin-y-0">
          {t<string[]>('setUpGrbReviewForm.documents.collapsableText.list', {
            returnObjects: true
          }).map(item => (
            <li key={item} className="margin-top-05">
              {item}
            </li>
          ))}
        </ul>
      </CollapsableLink>

      {grbReview.grbReviewType === SystemIntakeGRBReviewType.STANDARD && (
        <Alert slim type="info" className="margin-top-3">
          {t('setUpGrbReviewForm.documents.standardMeetingAlert')}
        </Alert>
      )}

      <h4 className="margin-top-6 margin-bottom-1">
        {t('setUpGrbReviewForm.documents.heading')}
      </h4>

      <Button
        type="button"
        onClick={() =>
          history.push(`/it-governance/${grbReview.id}/documents/upload`, {
            uploadSource: 'grbReviewForm'
          })
        }
      >
        {t('setUpGrbReviewForm.documents.addDocument')}
      </Button>

      <DocumentsTable
        systemIntakeId={grbReview.id}
        documents={grbReview.documents}
        className="margin-top-4"
      />
    </GRBReviewFormStepWrapper>
  );
};

export default AdditionalDocumentation;
