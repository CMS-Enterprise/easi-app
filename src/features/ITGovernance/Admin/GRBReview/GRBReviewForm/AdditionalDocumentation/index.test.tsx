import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { grbReview } from 'tests/mock/grbReview';

import { MessageProvider } from 'hooks/useMessage';

import AdditionalDocumentation from '.';

describe('GRB review form - additional documents', () => {
  it('renders the step', async () => {
    const { asFragment } = render(
      <MemoryRouter
        initialEntries={[`/it-governance/${grbReview.id}/grb-review/documents`]}
      >
        <MessageProvider>
          <Route path="/it-governance/:systemId/grb-review/:step">
            <AdditionalDocumentation grbReview={grbReview} />
          </Route>
        </MessageProvider>
      </MemoryRouter>
    );

    expect(
      screen.getByRole('heading', {
        name: 'Step 3 of 4 Additional documentation'
      })
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
