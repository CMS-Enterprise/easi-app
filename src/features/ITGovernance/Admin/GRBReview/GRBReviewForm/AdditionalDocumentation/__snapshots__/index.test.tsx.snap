// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GRB review form - additional documents > renders the step 1`] = `
<DocumentFragment>
  <main
    class="easi-main-content"
    id="main-content"
    role="main"
    tabindex="-1"
  >
    <div
      class="trb-form-header"
    >
      <div
        class="bg-gray-5 padding-bottom-6"
      >
        <div
          class="grid-container"
          data-testid="gridContainer"
        >
          <div
            class="grid-row"
            data-testid="grid"
          >
            <div
              class="grid-col"
              data-testid="grid"
            >
              <nav
                aria-label="Breadcrumbs"
                class="usa-breadcrumb usa-breadcrumb--wrap padding-bottom-0"
                style="background-color: transparent;"
              >
                <ol
                  class="usa-breadcrumb__list"
                >
                  <li
                    class="usa-breadcrumb__list-item"
                  >
                    <a
                      class="text-primary usa-breadcrumb__link"
                      href="/"
                    >
                      <span>
                        Home
                      </span>
                    </a>
                  </li>
                  <li
                    class="usa-breadcrumb__list-item"
                  >
                    <a
                      class="text-primary usa-breadcrumb__link"
                      href="/it-governance/a4158ad8-1236-4a55-9ad5-7e15a5d49de2/grb-review"
                    >
                      <span>
                        IT Governance request details
                      </span>
                    </a>
                  </li>
                  <li
                    aria-current="page"
                    class="usa-breadcrumb__list-item usa-current"
                  >
                    <span>
                      Set up GRB review
                    </span>
                  </li>
                </ol>
              </nav>
              <h1
                aria-live="polite"
                class="easi-h1 margin-bottom-0"
                tabindex="-1"
              >
                Set up GRB review
              </h1>
              <div
                class="font-body-lg line-height-body-5 text-light"
              >
                Use this form to set up a Governance Review Board (GRB) review in EASi.
              </div>
              <div
                class="margin-top-2 line-height-body-5"
              >
                You will be able to select the type of review, add recordings and documents, invite GRB members to review this project, and set a timeframe for an asynchronous review.
              </div>
              <div
                class="margin-top-3"
              >
                <button
                  class="usa-button usa-button--unstyled display-flex flex-align-center"
                  data-testid="button"
                  type="button"
                >
                  <svg
                    class="usa-icon margin-right-1"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                    />
                  </svg>
                  Save and return to request
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="grid-container"
        data-testid="gridContainer"
      />
      <div
        class="grid-container"
        data-testid="gridContainer"
      >
        <div
          aria-label="progress"
          class="usa-step-indicator margin-top-4 margin-bottom-2"
          data-testid="step-indicator"
        >
          <ol
            class="usa-step-indicator__segments"
          >
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-0"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Review type 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-1"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Presentation 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
            <li
              aria-current="true"
              aria-disabled="true"
              class="usa-step-indicator__segment usa-step-indicator__segment--current maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-2"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Additional documentation 
              </span>
            </li>
            <li
              aria-disabled="true"
              class="usa-step-indicator__segment maxw-none usa-step-indicator__segment--disabled"
              data-testid="stepIndicator-3"
            >
              <span
                class="usa-step-indicator__segment-label"
              >
                Participants and timeframe 
                <span
                  class="usa-sr-only"
                >
                  not completed
                </span>
              </span>
            </li>
          </ol>
          <div
            class="usa-step-indicator__header"
          >
            <h4
              class="usa-step-indicator__heading"
            >
              <span
                class="usa-step-indicator__heading-counter"
              >
                <span
                  class="usa-sr-only"
                >
                  Step
                </span>
                <span
                  class="usa-step-indicator__current-step"
                >
                  3
                </span>
                 
                <span
                  class="usa-step-indicator__total-steps"
                >
                  of 4
                </span>
                 
              </span>
              <span
                class="usa-step-indicator__heading-text"
              >
                Additional documentation
              </span>
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid-container grb-review-form-documents padding-bottom-10"
      data-testid="gridContainer"
    >
      <form
        class="usa-form step-content-wrapper maxw-none"
        data-testid="grbReviewForm-stepContentWrapper"
        role="form"
      >
        <p
          class="line-height-body-5 text-light font-body-sm margin-top-0 margin-bottom-1"
        >
          EASi will automatically link the Business Case (if applicable) and Intake Request form, but you may upload any additional documents necessary to provide GRB reviewers with project information or context. Documents already listed below were uploaded by the requester or project team as a part of their intake request. You may remove any documents uploaded by other Governance Admin Team members, but will not be able to remove documents uploaded by the requester or project team.
        </p>
        <div
          class="easi-collapsable-link"
        >
          <button
            aria-controls="documentsList"
            aria-expanded="false"
            class="usa-button usa-button--unstyled display-flex flex-align-center margin-top-3 padding-y-0"
            data-testid="collapsable-link"
            type="button"
          >
            <svg
              class="usa-icon margin-right-05"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            What documents should I upload?
          </button>
        </div>
        <div
          class="usa-alert usa-alert--info usa-alert--slim margin-top-3"
          data-testid="alert"
        >
          <div
            class="usa-alert__body"
          >
            <p
              class="usa-alert__text"
            >
              Because this review is a standard GRB meeting, GRB members may not review these documents unless they are presented during the meeting. However, adding additional documents here gives GRB members the option to access documentation within EASi prior to the meeting.
            </p>
          </div>
        </div>
        <h4
          class="margin-top-6 margin-bottom-1"
        >
          Documents
        </h4>
        <button
          class="usa-button"
          data-testid="button"
          type="button"
        >
          Add a document
        </button>
        <div
          class="margin-top-4"
          id="systemIntakeDocuments"
        >
          <div
            class="usa-table-container--scrollable"
            data-testid="scrollable-table-container"
            tabindex="0"
          >
            <table
              class="usa-table usa-table--borderless width-full"
              data-testid="table"
            >
              <thead>
                <tr
                  role="row"
                >
                  <th
                    aria-sort="none"
                    class="border-bottom-2px"
                    colspan="1"
                    role="columnheader"
                    scope="col"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <button
                      class="usa-button usa-button--unstyled"
                      data-testid="button"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                      type="button"
                    >
                      File name
                      <svg
                        class="usa-icon margin-left-05"
                        data-testid="caret--sort"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                        />
                      </svg>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="border-bottom-2px"
                    colspan="1"
                    role="columnheader"
                    scope="col"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <button
                      class="usa-button usa-button--unstyled"
                      data-testid="button"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                      type="button"
                    >
                      Document type
                      <svg
                        class="usa-icon margin-left-05"
                        data-testid="caret--sort"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                        />
                      </svg>
                    </button>
                  </th>
                  <th
                    aria-sort="descending"
                    class="border-bottom-2px"
                    colspan="1"
                    role="columnheader"
                    scope="col"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <button
                      class="usa-button usa-button--unstyled"
                      data-testid="button"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                      type="button"
                    >
                      Date added
                      <svg
                        class="usa-icon margin-left-05"
                        data-testid="caret--down"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </button>
                  </th>
                  <th
                    aria-sort="none"
                    class="border-bottom-2px"
                    colspan="1"
                    role="columnheader"
                    scope="col"
                    style="cursor: pointer;"
                    title="Toggle SortBy"
                  >
                    <button
                      class="usa-button usa-button--unstyled"
                      data-testid="button"
                      style="cursor: pointer;"
                      title="Toggle SortBy"
                      type="button"
                    >
                      Actions
                      <svg
                        class="usa-icon margin-left-05"
                        data-testid="caret--sort"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
                        />
                      </svg>
                    </button>
                  </th>
                </tr>
              </thead>
              <tbody
                role="rowgroup"
              >
                <tr
                  role="row"
                >
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    sample3.pdf
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    <p>
                      Some other type of doc
                    </p>
                    <p
                      class="text-base font-sans-2xs"
                    >
                      Historical
                    </p>
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    06/14/2023
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    Unavailable
                  </td>
                </tr>
                <tr
                  role="row"
                >
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    sample1.pdf
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    <p>
                      Draft IGCE
                    </p>
                    <p
                      class="text-base font-sans-2xs"
                    >
                      Current
                    </p>
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    06/14/2023
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    <div
                      class="display-flex flex-row"
                    >
                      <button
                        class="usa-button usa-button--unstyled"
                        data-testid="button"
                        type="button"
                      >
                        Download
                      </button>
                      <button
                        class="usa-button usa-button--unstyled margin-left-2 text-error"
                        data-testid="button"
                        type="button"
                      >
                        Remove
                      </button>
                    </div>
                  </td>
                </tr>
                <tr
                  role="row"
                >
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    sample2.pdf
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    <p>
                      SOO, SOW, PWS, or other contracting document
                    </p>
                    <p
                      class="text-base font-sans-2xs"
                    >
                      Current
                    </p>
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    06/14/2023
                  </td>
                  <td
                    class="text-ttop"
                    role="cell"
                  >
                    <em
                      data-testurl="http://localhost:9004/easi-app-file-uploads/7e047111-6228-4943-9c4b-0961f27858f4.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minioadmin%2F20230614%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20230614T184943Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=0e3f337697c616b01533accd95a316cbeabeb6990961b9881911c757837cbf95"
                    >
                      Virus scan in progress...
                    </em>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
        <div
          class="margin-top-8 margin-bottom-3"
        >
          <ul
            class="usa-button-group"
          >
            <li
              class="usa-button-group__item"
            >
              <div
                class="display-flex flex-align-center"
              >
                <button
                  class="usa-button usa-button--outline margin-top-0 margin-bottom-1 mobile-lg:margin-bottom-0"
                  data-testid="button"
                  type="button"
                >
                  Back
                </button>
              </div>
            </li>
            <li
              class="usa-button-group__item"
            >
              <div
                class="display-flex flex-align-center"
              >
                <button
                  class="usa-button margin-top-0"
                  data-testid="button"
                  disabled=""
                  type="submit"
                >
                  Next
                </button>
              </div>
            </li>
          </ul>
          <button
            class="usa-button usa-button--unstyled margin-top-2 display-flex flex-align-center"
            data-testid="button"
            type="button"
          >
            <svg
              class="usa-icon margin-right-05"
              focusable="false"
              height="1em"
              role="img"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
            Save and return to request
          </button>
        </div>
      </form>
    </div>
  </main>
</DocumentFragment>
`;
